go/vt/discovery/replicationlag.go

```diff
@@ -25,9 +25,10 @@ import (
 
 var (
 	// lowReplicationLag defines the duration that replication lag is low enough that the VTTablet is considered healthy.
-	lowReplicationLag            = flag.Duration("discovery_low_replication_lag", 30*time.Second, "the replication lag that is considered low enough to be healthy")
-	highReplicationLagMinServing = flag.Duration("discovery_high_replication_lag_minimum_serving", 2*time.Hour, "the replication lag that is considered too high when selecting the minimum num vttablets for serving")
-	minNumTablets                = flag.Int("min_number_serving_vttablets", 2, "the minimum number of vttablets that will be continue to be used even with low replication lag")
+	lowReplicationLag             = flag.Duration("discovery_low_replication_lag", 30*time.Second, "the replication lag that is considered low enough to be healthy")
+	highReplicationLagMinServing  = flag.Duration("discovery_high_replication_lag_minimum_serving", 2*time.Hour, "the replication lag that is considered too high when selecting the minimum num vttablets for serving")
+	minNumTablets                 = flag.Int("min_number_serving_vttablets", 2, "the minimum number of vttablets that will be continue to be used even with low replication lag")
+	legacyReplicationLagAlgorithm = flag.Bool("legacy_replication_lag_algorithm", false, "use the legacy algorithm when selecting the vttablets for serving")
```

Comment:
I think this should initially default to true. Otherwise behavior would change for existent users without them opting in. 

We can then go through a process of:
1) Opt in to new behavior.
2) New behavior becomes default.
3) Deprecate legacy behavior.

Reply:
It is now default to true.